Profile/aw/54434/add mobile edit form to page

adamwhitlock1 opened the pull request:
## Summary

- Renders a mobile phone editing page complete with form and 'auto-return' navigation on save success.
- When running the mock server, you can test this by visiting `profile/notifications` and seeing the alert that no mobile number is present on the user's profile. 
  - You can then click the link in the alert, visit the edit page, put in a valid phone number, and save the form to see the save action complete and the user will then be returned to the notifications page and the notifications page will now show all the notifications.
  - The notifications page will also show a success alert with a checkmark that alerts the user that their mobile phone number info has been updated successfully.
  - This is not a persistent save because of the way the mock server works, so if you refresh the notifications page again, you will see that the alert re-appears as if the user has no mobile number again. Once this PR is merged, then we can validate this further as an E2E flow with any staging user.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/54434

## Testing done

- Unit tests created for basic coverage on this. Going to expand on the tests and get some E2E tests built out in the next ticket, which will mostly be cleanup and testing focused.
- I put some steps in the description on how to mess around with this page and test it.


## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

No real before/after shots since this is a new page, but i have some screenshots of the new form and success alert.

Form for mobile number:
<img width="1058" alt="Screenshot 2023-04-18 at 5 49 23 PM" src="https://user-images.githubusercontent.com/8332986/232928866-9effc2b4-31a5-45c4-9399-6f2ac82b8c2d.png">

Alert for save success:
<img width="1013" alt="Screenshot 2023-04-18 at 5 49 48 PM" src="https://user-images.githubusercontent.com/8332986/232928946-c6f7d520-18b0-4c8e-b7a4-a04c6a628f26.png">

## What areas of the site does it impact?
Profile - behind feature toggle only

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  I added a screenshot of the developed feature
- [x]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

Commit:
feat: add field controller to edit page

adamwhitlock1 committed:
fix: use default export instead of named controller smh

adamwhitlock1 committed:
feat: custom save and cancel btn text support

adamwhitlock1 committed:
feat: edit page with redirect and update aler

adamwhitlock1 committed:
test: unit test edit page

adamwhitlock1 committed:
Merge branch 'main' into Profile/aw/54434/add-mobile-edit-form-to-page

adamwhitlock1 committed:
Merge branch 'main' into Profile/aw/54434/add-mobile-edit-form-to-page

adamwhitlock1 committed:
Merge branch 'main' into Profile/aw/54434/add-mobile-edit-form-to-page

Commit:
fix: hide extension field and label for mobile

andaleliz commented:
Hi @adamwhitlock1, yep, I think that's fun to use the word "number" in the heading! Thanks for checking with me. This is looking great!

allisonlu commented on src/applications/personalization/profile/components/ProfileInformationEditView.jsx:
minor ES lint errors for import paths here... should be fixed by replacing `platform/` with `~/platform/`

adamwhitlock1 replied:
Oh yea, Ill clean those pesky warnings up. Thanks

allisonlu commented on src/applications/personalization/profile/components/alerts/FieldHasBeenUpdated.jsx:
```diff
@@ -0,0 +1,22 @@
+      class="vads-u-margin-bottom--1"
+      background-only
+      status="success"
+      show-icon
```
this is different from the Sketch file, but shouldn't be a big deal since you're using a design system component

adamwhitlock1 replied:
yep, I mentioned that to Liz as well and she said she will update the sketch file.

allisonlu commented on src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:
```diff
@@ -91,21 +92,22 @@ const NotificationSettings = ({
     <>
       <Headline>{PROFILE_PATH_NAMES.NOTIFICATION_SETTINGS}</Headline>
-      {shouldShowLoadingIndicator ? (
+      {shouldShowLoadingIndicator && (
```
this operator is so useful 😩

allisonlu commented on src/platform/user/profile/vap-svc/components/ProfileInformationFieldController.jsx:
a few ES lint errors here too

allisonlu commented on src/applications/personalization/profile/mocks/endpoints/status.js:
```diff
@@ -67,7 +67,7 @@ module.exports = {
       type: 'async_transaction_va_profile_address_transactions',
       attributes: {
-        transactionId: '89747e43-d01b-4024-81fb-5e0361f9cf9f',
+        transactionId: 'success-test-id',
```
just curious - where is `success-test-id` coming from?

adamwhitlock1 replied:
It came from my brain lol. IDK I created a 'named' id here so that I could differentiate the different mock transactionsIds easier and not have a rando id that I would have to look at to determine from.

allisonlu commented:
minor discrepancies with the Sketch file:
- (*Required) text is appear next to help text label instead of a new line
- the input field width is wider than the one in the Sketch file
- made a note above: the check mark icon is slightly different

other than that, e2e and unit tests look good and all are passing 👍

adamwhitlock1 committed:
fix: eslint warnings

adamwhitlock1 commented:
fixed up the eslint warnings. Had a small issue with the recordEvent, but found other instances importing it and having to disable eslint for that import so thats what I did too.

adamwhitlock1 commented:
@allisonlu I updated the eslint warning stuff and cleaned those imports up. I think the slight design differences should be okay, since they match the existing layout that is within main contact information page. I attached some screenshots for comparison.

EDIT PAGE
![Screenshot 2023-04-20 at 1 14 04 PM](https://user-images.githubusercontent.com/8332986/233465103-5c5a36d6-fead-4d54-ac42-d7f31cbe2b22.png)

CONTACT INFO PAGE
![Screenshot 2023-04-20 at 1 13 48 PM](https://user-images.githubusercontent.com/8332986/233465137-93b1dc7d-56fd-4961-89b6-7f4eac9eabd8.png)

allisonlu reviewed (approved):
lgtm!
